source/D2Common/include/Drlg/D2DrlgRoomTile.h

```diff
@@ -5,9 +5,20 @@
 
 #pragma pack(1)
 
+enum D2C_PackedTileInformationMasks
+{
+	PACKEDTILEINFO_ROOF = 0b00,
+	PACKEDTILEINFO_WALL = 0b01,
+	PACKEDTILEINFO_FLOOR = 0b10,
+	PACKEDTILEINFO_STRUCTUREPARTMASK = PACKEDTILEINFO_ROOF | PACKEDTILEINFO_WALL | PACKEDTILEINFO_FLOOR,
+	PACKEDTILEINFO_ANIMATEDTILE = 1 << 27,
+};
 
 #pragma pack()
 
+// Helper functions
+static uint32_t GetTileStyleFromPackedTileInformation(uint32_t nTileInformation) { return (nTileInformation >> 20) & 0x3F; }
+static uint32_t GetTileSequenceFromPackedTileInformation(uint32_t nTileInformation) { return BYTE1(nTileInformation); }
```

Comment:
Theoretically, these could be used at way more places than they are currently

Reply:
I think so too.

Mentor's note have the following table, which I need to confirm but seems to match.


<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Lectem/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Lectem/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:1.0in .75in 1.0in .75in;
	mso-header-margin:.5in;
	mso-footer-margin:.5in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:windowtext;
	font-size:10.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:Arial;
	mso-generic-font-family:auto;
	mso-font-charset:238;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
.xl65
	{font-size:8.0pt;
	font-family:Arial, sans-serif;
	mso-font-charset:0;
	text-align:left;}
.xl66
	{font-size:8.0pt;
	font-weight:700;
	font-family:Arial, sans-serif;
	mso-font-charset:0;
	text-align:left;}
.xl67
	{font-size:8.0pt;
	font-family:Arial, sans-serif;
	mso-font-charset:0;
	text-align:left;
	white-space:normal;}
.xl68
	{font-size:8.0pt;
	font-family:Arial, sans-serif;
	mso-font-charset:0;
	text-align:left;
	border:.5pt solid windowtext;}
.xl69
	{font-size:8.0pt;
	font-weight:700;
	font-family:Arial, sans-serif;
	mso-font-charset:0;
	text-align:left;
	border:.5pt solid windowtext;}
.xl70
	{font-size:8.0pt;
	font-family:Arial, sans-serif;
	mso-font-charset:0;
	text-align:left;
	border:.5pt solid windowtext;
	white-space:normal;}
.xl71
	{font-size:8.0pt;
	font-family:Arial, sans-serif;
	mso-font-charset:0;
	mso-number-format:"\@";
	text-align:left;}
.xl72
	{color:red;
	font-size:8.0pt;
	font-weight:700;
	text-decoration:underline;
	text-underline-style:single;
	font-family:Arial, sans-serif;
	mso-font-charset:0;
	text-align:left;
	border:.5pt solid windowtext;}
.xl73
	{font-size:8.0pt;
	font-weight:700;
	font-family:Arial, sans-serif;
	mso-font-charset:0;
	text-align:left;
	border:.5pt solid windowtext;
	white-space:normal;}
.xl74
	{font-size:8.0pt;
	font-weight:700;
	font-family:Arial, sans-serif;
	mso-font-charset:0;
	mso-number-format:"\@";
	text-align:left;
	border:.5pt solid windowtext;}
-->
</style>
</head>

<body link=blue vlink="#954F72">



flag | bit | enum eDS1Flags | comment | tiledata flag
-- | -- | -- | -- | --
0x00000001 | 0 | TILEFLAG_WALL | default   valid wall |  
0x00000002 | 1 | TILEFLAG_FLOOR | default   valid floor | 0x2000
0x00000004 | 2 | TILEFLAG_LOS | code-generated;   added on edges |  
0x00000008 | 3 | TILEFLAG_ENCLOSE | seems   to delimit an enclosure inside another area; | 0x04
0x00000010 | 4 | TILEFLAG_EXIT | arch   or doorway or gateway in a wall |  
0x00000020 | 5 | TILEFLAG_20 |   |  
0x00000040 | 6 | TILEFLAG_LAYER_BELOW | only   floor |  
0x00000080 | 7 | TILEFLAG_LAYER_ABOVE | wall   & floor | 0x01
0x00000100 | 8 | TILEFLAG_TILESUBINDEX |   |  
0x00000200 | 9 | TILEFLAG_TILESUBINDEX |   |  
0x00000400 | 10 | TILEFLAG_TILESUBINDEX |   |  
0x00000800 | 11 | TILEFLAG_TILESUBINDEX |   |  
0x00001000 | 12 | TILEFLAG_TILESUBINDEX |   |  
0x00002000 | 13 | TILEFLAG_TILESUBINDEX |   |  
0x00004000 | 14 | TILEFLAG_TILESUBINDEX | seems   to be subindex, too. AND 0x3F0FF00 is used to check if index 0 subindex 0 |  
0x00008000 | 15 | TILEFLAG_TILESUBINDEX | seems   to be subindex, too. AND 0x3F0FF00 is used to check if index 0 subindex 0 |  
0x00010000 | 16 | TILEFLAG_FILL_LOS | all   subtiles will get wall collision; | 0x80
0x00020000 | 17 | TILEFLAG_UNWALKABLE |   | 0x40
0x00040000 | 18 | TILEFLAG_OVERLAPPED_LAYER2 | code-generated;   set when at least Floor2 or Wall2  //   (nLayer - 1) << 0x12 |  
0x00080000 | 19 | TILEFLAG_OVERLAPPED_LAYER3 | code-generated;   set when has more walls, incl' of other orientations than usual; obj, shd,   tree, roof, lower |  
0x000C0000 | 19 | TILEFLAG_OVERLAPPED_LAYER4 | code-generated;   set when has more walls, incl' of other orientations than usual; obj, shd,   tree, roof, lower |  
0x00100000 | 20 | TILEFLAG_TILEINDEX | SHR   R32, 14  //  AND R32, 3F |  
0x00200000 | 21 | TILEFLAG_TILEINDEX |   |  
0x00400000 | 22 | TILEFLAG_TILEINDEX |   |  
0x00800000 | 23 | TILEFLAG_TILEINDEX |   |  
0x01000000 | 24 | TILEFLAG_TILEINDEX |   |  
0x02000000 | 25 | TILEFLAG_TILEINDEX |   |  
0x04000000 | 26 | TILEFLAG_REVEAL_HIDDEN | looks   like an upper wall brought to a layer in front | 0x20C
0x08000000 | 27 | TILEFLAG_SHADOW | this   layer is a shadow layer |  
0x10000000 | 28 | TILEFLAG_LINKAGE | near   wp, lvl links, paths // will never get hidden | 0x102
0x20000000 | 29 | TILEFLAG_OBJECT_WALL | wall   tiles with props; may be block reverb / other sounds (crates, barrels, tables   etc.) | 0x800
0x40000000 | 30 | TILEFLAG_40000000 |   |  
0x80000000 | 31 | TILEFLAG_HIDDEN |   | 0x08



</body>

</html>

Reply:
I will investigate this in a future PR